src/Neo/SmartContract/Native/ContractManagement.cs

```diff
@@ -143,6 +146,39 @@ public ContractState GetContract(DataCache snapshot, UInt160 hash)
             return snapshot.TryGet(CreateStorageKey(Prefix_Contract).Add(hash))?.GetInteroperable<ContractState>();
         }
 
+        /// <summary>
+        /// Maps specified ID to deployed contract.
+        /// </summary>
+        /// <param name="snapshot">The snapshot used to read data.</param>
+        /// <param name="id">Contract ID.</param>
+        /// <returns>The deployed contract.</returns>
+        [ContractMethod(CpuFee = 1 << 15, RequiredCallFlags = CallFlags.ReadStates)]
+        public ContractState GetContractById(DataCache snapshot, int id)
```

Comment:
```suggestion
        public ContractState GetContractById(DataCache snapshot, uint id)
```

Reply:
Seems this is only for non-native contracts, `uint` is better.

Reply:
I think it's `GetContractHashes` that is specifically for non-natives, but this one can return native contracts as well, it doesn't hurt. Otherwise we'll have to check for int overflows anyway.